runtime/src/main/cpp/Natives.cpp

```diff
@@ -47,6 +47,10 @@ OBJ_GETTER(Kotlin_getStackTraceStrings, KConstRef stackTrace) {
   RETURN_RESULT_OF(GetStackTraceStrings, stackTrace);
 }
 
+KLong Kotlin_getImageBase(KNativePtr address) {
+  return static_cast<KLong>(konan::getImageBase(address));
```

Comment:
I guess this line sign-extends `address` to `KLong`. Is it correct on 32-bit platforms?

Reply:
Right.